src/client.rs

```diff
@@ -326,10 +326,10 @@ where
     /// The response headers are stored in the provided rx_buf, which should be sized to contain at least the response headers.
     ///
     /// The response is returned.
-    pub async fn send<'buf>(
-        &'buf mut self,
+    pub async fn send<'buf, 'req>(
```

Comment:
Can we swap the lifetime arguments here to align with the other send functions. We have throughout tried to have the convention that the longest lifetimes "comes first"